pkg/cloud/packet/client.go

```diff
@@ -303,23 +322,25 @@ func (p *Client) EnsureNodeBGPEnabled(id string) error {
 	return err
 }
 
-func (p *Client) GetIPByClusterIdentifier(namespace, name, projectID string) (packngo.IPAddressReservation, error) {
+func (p *Client) GetIPByClusterIdentifier(ctx context.Context, namespace, name, projectID string) (*metal.IPReservation, error) {
 	var err error
-	var reservedIP packngo.IPAddressReservation
+	var ipReservation *metal.IPReservation
 
-	listOpts := &packngo.ListOptions{}
-	reservedIPs, _, err := p.ProjectIPs.List(projectID, listOpts)
+	reservedIPs, _, err := p.IPAddressesApi.FindIPReservations(ctx, projectID).Execute()
 	if err != nil {
-		return reservedIP, err
+		return ipReservation, err
 	}
-	for _, reservedIP := range reservedIPs {
-		for _, v := range reservedIP.Tags {
-			if v == generateElasticIPIdentifier(name) {
-				return reservedIP, nil
+	for _, reservedIPWrapper := range reservedIPs.IpAddresses {
+		ipReservation = reservedIPWrapper.IPReservation
+		if ipReservation != nil {
```

Comment:
Since we're creating a plain old `IPReservation` in `CreateIP` above, I _think_ it's reasonable here to ignore `VrfIpReservations`.

Reply:
If `ipReservation.Tags` below were `GetTags()` I think we could ignore the nil check and enhance our apathy about the reserved IP type.

Reply:
Using `GetTags()` doesn't help here because `ipReservation` is nil, so we get an NPE from attempting to access a property of something that is nil.